PactSafe iOS Demo/Controllers/LoginViewController.swift

```diff
@@ -6,158 +6,154 @@
+
+    if signerId.isEmpty {
+      let alert = basicAlert("Error", message: "Missing Email Address. Please try again.")
+      present(alert, animated: true, completion: nil)
+    }
+
+    if passwordText.isEmpty {
+      let alert = basicAlert("Error", message: "Missing Password. Please try again.")
+      present(alert, animated: true, completion: nil)
+    }
+
+    /// Check the status of the specified signer ID within a PactSafe group.
+    ps.signedStatus(for: signerId, groupKey: groupKey) { needsAcceptance, contractIds in
+      if needsAcceptance {
+        self.showContractUpdates(forSignerId: signerId,
+                                 password: passwordText,
+                                 filterContractIds: contractIds)
+      } else {
+        DispatchQueue.main.async {
+          self.segueToHome()
         }
+      }
     }
-    
-    private func showContractUpdates(forSignerId signerId: String,
-                                     password passwordText: String,
-                                     filterContractIds: [String]? = nil) {
-        
-        self.ps.loadGroup(groupKey: self.groupKey) { (groupData, error) in
-            guard let groupData = groupData, let contractsData = groupData.contractData else { return }
-            self.psGroupData = groupData
-            
-            var titlesOfContracts = [String]()
-            
-            var updatedContractsMessage: String = "We've updated the following: "
-            
-            if let cidsFilter = filterContractIds {
-                contractsData.forEach { (key, value) in
-                    if cidsFilter.contains(key) { titlesOfContracts.append(value.title) }
-                }
-            } else {
-                contractsData.forEach { (key, value) in
-                    titlesOfContracts.append(value.title)
-                }
-            }
-            
-            let contractTitles = titlesOfContracts.map { String($0) }.joined(separator: " and ")
-            updatedContractsMessage.append(contractTitles)
-            updatedContractsMessage.append(".\n \n Please agree to these changes.")
-            
-            DispatchQueue.main.async {
-                let alert = self.updatedTermsAlert("Updated Terms", message: updatedContractsMessage, email: signerId, password: passwordText)
-                self.present(alert, animated: true, completion: nil)
-            }
+  }
+
+  private func showContractUpdates(forSignerId signerId: String,
+                                   password passwordText: String,
+                                   filterContractIds: [String]? = nil) {
+    ps.loadGroup(groupKey: groupKey) { groupData, _ in
+      guard let groupData = groupData, let contractsData = groupData.contractData else { return }
+      self.psGroupData = groupData
+
+      var titlesOfContracts = [String]()
+
+      var updatedContractsMessage: String = "We've updated the following: "
+
+      if let cidsFilter = filterContractIds {
```

Comment:
can you explain what is happening here? this is syntactically strange haha

Reply:
Basically, if the user decides to pass an array of contract IDs...then we will unwrap the optional and filter the list of contract titles to only be those. If they choose not to pass, we will then add all contract titles to the `titleOfContracts` array. This could definitely be cleaner and use more modern syntax with `.map` or something similar. The demo app needs to be redone to be more modern since Swift has been swiftly evolving each year 😎

Reply:
that syntax is cray. it uses the newly defined `cidsFilter` inside of that block? so funny. And np, I was just curious!